### ISSUE TITLE
basename: invalid option --'a' issue on Centos 6.7

### ISSUE THREAD

**Issue by egidijusz** [oh-my-fish/oh-my-fish#179]
Note: workaround is at the bottom of this comment.

When doing `omf theme`, `omf update`, `omf describe` and etc. I get following output (and command is not executed):
```
⋊> ~ omf theme                                                                                                                                                                                               23:05:26
basename: invalid option -- 'a'
Try `basename --help' for more information.

```

Same output (invalid option -- 'a') was while installing omf itself:
```
~> curl -L https://github.com/oh-my-fish/oh-my-fish/raw/master/bin/install | fish
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
101  4654  101  4654    0     0   7156      0 --:--:-- --:--:-- --:--:--  7156
Installing Oh My Fish to /home/egis/.local/share/omf...
Cloning master from https://github.com/oh-my-fish/oh-my-fish.git...
Adding startup code to fish config file...
Building Oh My Fish configuration...
basename: invalid option -- 'a'
Try `basename --help' for more information.
Installing theme default
✔ theme default successfully installed.
Theme not installed!
Install it using omf install default
Welcome to fish, the friendly interactive shell
Type help for instructions on how to use fish
⋊> ~                                                       
```

Additional info.
```
⋊> ~ cat /etc/centos-release                                                                                                                                                                                 23:08:28
CentOS release 6.7 (Final)

```
```
⋊> ~ fish --version                                                                                                                                                                                          23:09:38
fish, version 2.2.0

```
```
⋊> ~ basename --version                                                                                                                                                                                      23:10:07
basename (GNU coreutils) 8.4
Copyright (C) 2010 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>.
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.

Written by David MacKenzie.

```

**Workaround**
Upgrade `basename` to newer version, add dir with upgraded `basename` to your $PATH, install ohmyfish.

I used this http://unix.stackexchange.com/questions/205397/upgrade-chroot-version-on-centos-5 as a reference for upgrading basename.
Basically you have to dowload GNU Coreutils source and build it using `./configure --prefix=/path/to/your/custom/directory". /path/to/your/cutom/directory is important - you don't want to break things. Then do `make && make install`.
Do 
```
echo 'set PATH /path/to/your/custom/directory $PATH' >> ~/.config/fish/config.fish
``` 
This adds new basename to your $PATH so omf can use it.

Install omf.

Sidenote - omf will replace your config.fish file, so you'll need to re-add $PATH line after installation of omf.

**Comment by egidijusz** [comment]
P.s. installing `basename-compat` plugin did not help (still got the same 'invalid option` error).

**Comment by derekstavis** [comment]
For this cases the appropriate thing to do is to keep system up to date. For `basename-compat`, it probably should handle `-a` option.

**Comment by bpinto** [comment]
@derekstavis Maybe the plugin is loaded only after it could be useful?

**Comment by derekstavis** [comment]
Yeah, this is a real issue. Anyway, using an outdated coreutils is very dangerous.

**Comment by agarzon** [comment]
Im having the exact same issue:

```
root@vps ~# fish install
Installing Oh My Fish to /root/.local/share/omf...
Cloning master from https://github.com/oh-my-fish/oh-my-fish.git...
Existent config.fish found at /root/.config/fish
↳ Moving to /root/.config/fish/config.1448503561.copy
Adding startup code to fish config file...
Building Oh My Fish configuration...
basename: invalid option -- 'a'
Try `basename --help' for more information.
Installing theme default
Unpacking objects: 100% (18/18), done.
✔ theme default successfully installed.
Theme not installed!
Install it using omf install default
```

**Comment by derekstavis** [comment]
@agarzon Please update your distribution's coreutils.
